### PR TITLE
Show and clear only predefined LOIs

### LINKED ISSUE #1570: [LOI editor] Implementation incomplete
- [x] "Clear all" should only delete predefined LOIs (without `planned: false`, i.e., if `null` or `true`)
- [x]  Map should only show predefined LOIs
- [x] Big scary warning should be shown when clicking "Clear all". @amegantz for copy, something like:
   > Clear predefined sites
   > This action will delete all predefined data collection sites. All associated data will be lost. Are you sure?

@rfontanarosa FYI

### PR DESCRIPTION
**Pull request by rfontanarosa**
part of #1570 

https://github.com/google/ground-platform/assets/10047945/870170ba-934e-4cc5-8924-bce3c34f4a23

### CODE REVIEW

File: web/src/app/pages/create-survey/survey-loi/survey-loi.component.ts [hunk 1]
```diff
@@ -43,7 +43,11 @@ export class SurveyLoiComponent {
   ) {}
 
   async ngOnInit() {
-    this.lois$ = this.loiService.getLocationsOfInterest$();
+    this.loiService
+      .getLocationsOfInterest$()
```
Comment (gino-m) [inline web/src/app/pages/create-survey/survey-loi/survey-loi.component.ts:47]: Can we reuse the logic in EditJob, perhaps refactor it into the loiService?

I.e.:
```
function getPredefinedLoisByJobId(jobId) {
}
```

**Review by gino-m — APPROVED** [review]
LGTM, one optional suggestion.

File: web/src/app/pages/create-survey/survey-loi/survey-loi.component.ts (hunk 1 above)
Comment (rfontanarosa) [reply web/src/app/pages/create-survey/survey-loi/survey-loi.component.ts:47]: yes

**Review by gino-m — APPROVED** [review]

File: web/src/app/pages/create-survey/survey-loi/survey-loi.component.ts (hunk 1 above)
Comment (gino-m) [reply web/src/app/pages/create-survey/survey-loi/survey-loi.component.ts:47]: Great. Will you do it?

File: web/src/app/components/loi-editor/loi-editor.component.html [hunk 2]
```diff
@@ -19,35 +19,35 @@
...
 
         <button *ngIf="lois.size > 0"
-          mat-button
+          mat-flat-button
           class="clear-all-lois"
-          (click)="clearLois(survey.id, lois)"
-          color="primary"
+          (click)="clearLois()"
         >
+          <mat-icon class="material-symbols-outlined">delete</mat-icon>
```
Comment (gino-m) [inline web/src/app/components/loi-editor/loi-editor.component.html:44]: Can we also change this label to "Delete all" as suggested by @amegantz in https://github.com/google/ground-platform/issues/1570#issuecomment-1986547600 ?

**Review by gino-m — CHANGES_REQUESTED** [review]

**Review by gino-m — APPROVED** [review]
LGTM!

**Pull request merged**